### PR TITLE
Allow metrics input to be a str for AutoMMPredictor evaluation

### PR DESCRIPTION
**Pull request by FANGAreNotGnu**
#1844 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: multimodal/src/autogluon/multimodal/predictor.py [hunk 1]
```diff
@@ -1345,7 +1345,7 @@ def _logits_to_prob(logits: torch.Tensor):
     def evaluate(
         self,
         data: Union[pd.DataFrame, dict, list],
-        metrics: Optional[List[str]] = None,
+        metrics: Optional[str or List[str]] = None,
```
Comment (sxjscience) [inline multimodal/src/autogluon/multimodal/predictor.py:1348]: Use `Optional[Union[str, List[str]]]`. See https://docs.python.org/3/library/typing.html#typing.Union

File: multimodal/src/autogluon/multimodal/predictor.py [hunk 2]
```diff
@@ -1403,6 +1403,8 @@ def evaluate(
 
         if metrics is None:
             metrics = [self._eval_metric_name]
+        if type(metrics) is not list:
```
Comment (sxjscience) [inline multimodal/src/autogluon/multimodal/predictor.py:1406]: Use `isinstance` instead.

For example:

```python
if isinstance(metrics, str):
   metrics = [metrics]
```

**Comment by szha** [comment]
Job PR-1899-1 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1899/1/index.html

**Comment by szha** [comment]
Job PR-1899-2 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1899/2/index.html

**Review by sxjscience — APPROVED** [review]

**Pull request merged**